pkg/server/server_integration_test.go

```diff
@@ -37,15 +37,50 @@ func TestSendingMessagesBetweenPeers(t *testing.T) {
 	ws2 := peerJoinsRoom(t, httpServer, room, peerSecret2)
 	defer ws2.Close()
 
+	readMessage(t, ws1) // skip 'peer_connected'
+
 	m1 := agent.ClientMessage{To: peer2, Payload: json.RawMessage(`"ping"`)}
 	sendMessage(t, ws1, m1)
 	recv1 := readMessage(t, ws2)
-	assertSameMessages(t, peer1, m1, recv1)
+	assertSameClientMessages(t, peer1, m1, recv1)
 
 	m2 := agent.ClientMessage{To: peer1, Payload: json.RawMessage(`"pong"`)}
 	sendMessage(t, ws2, m2)
 	recv2 := readMessage(t, ws1)
-	assertSameMessages(t, peer2, m2, recv2)
+	assertSameClientMessages(t, peer2, m2, recv2)
+}
+
+func TestSendingControlMessages(t *testing.T) {
+	store := messaging.NewRoomStore()
```

Comment:
I don;'t think it deserves an integration test. I'd just do it in the agent tests. There it should be simpler, just use mocked broker and assert on the messages sent to the broker when agent connects and disconnects.